### PR TITLE
feat: failure on warning

### PR DESCRIPTION
**Pull request by thecampagnards**
Hello,
I suggest this pr to `exit 1` when a warning appears.
I added a flag `--failure-on-warning` or `-f` to activate it when you want.
We need it to force the failure of our CI in case of warnings.
Thx!

### CODE REVIEW

**Comment by thecampagnards** [comment]
> Great addition. Does there exist prior art, i.e. does [eslint](https://eslint.org/) offer a similar flag?

Yes they have ˋ --max-warnings’ https://eslint.org/docs/latest/user-guide/command-line-interface#max-warnings I can change to that

**Comment by nikku** [comment]
Please change to that. Also enhances the feature. :+1:

**Comment by thecampagnards** [comment]
updated + fixed lint issues

File: bin/bpmnlint.js [hunk 1]
```diff
@@ -334,9 +337,10 @@ async function run() {
     init,
     version,
     config: configOverridePath,
+    'max-warnings': maxWarningsOverride,
     _: files
   } = mri(process.argv.slice(2), {
-    string: [ 'config' ],
+    string: [ 'config', 'max-warnings' ],
```
Comment (nikku) [inline bin/bpmnlint.js:343]: Isn't it a number?

Comment (nikku) [reply bin/bpmnlint.js:343]: We should provide a default for it and get rid of the overriding code: https://www.npmjs.com/package/mri#user-content-optionsdefault.

**Review by nikku — CHANGES_REQUESTED** [review]
We should add an [integration test](https://github.com/bpmn-io/bpmnlint/blob/master/test/integration/cli-spec.js#L66) to verify the feature works.

File: bin/bpmnlint.js (hunk 1 above)
Comment (thecampagnards) [reply bin/bpmnlint.js]: > Isn't it a number?

there is no number option https://github.com/lukeed/mri/blob/master/index.d.ts#L8, `Numerical values are cast as Numbers when possible`

**Comment by nikku** [comment]
Fixed + incorporated some nit-picks into https://github.com/bpmn-io/bpmnlint/pull/92.

**Pull request closed without merging**